### PR TITLE
Adds release notes to documentation

### PR DESCRIPTION
**Pull request by josh146**
This PR adds the release notes to the documentation.

Screenshot_20200113_185522

This is simply a rendering of the existing `.github/CHANGELOG.md` file.

Minimal changes were made where possible; these include:

* The `m2r` extension is used to include the changelog, using the `.. mdinclude::` directive.

* Subheadings were changed to use `<h3>Subheading</h3>` to avoid them showing up in the table of contents. This will continue to render properly on github.

* A few small code examples were added for the latest development release

* Links to PRs were modified to add parenthesis around the PR number. This makes them look less 'odd'.

### CODE REVIEW

**Comment by mariaschuld** [comment]
I'd like to build it before accepting, but get the following error:

Configuration error: 
There is a programmable error in your configuration file: 
...
ModuleNotFoundError: No module named 'pennylane_qchem'

File: doc/conf.py [hunk 1]
```diff
@@ -45,8 +45,11 @@
     "sphinx.ext.intersphinx",
     "sphinx_automodapi.automodapi",
     'sphinx_copybutton',
+    "m2r"
```
Comment (mariaschuld) [inline doc/conf.py]: I needed to install m2r to build the docs. Do we need to add it to requirements?

File: doc/index.rst [hunk 2]
```diff
@@ -184,6 +184,7 @@ PennyLane is **free** and **open source**, released under the Apache License, Ve
    development/guide
    development/plugins
    development/research
+   development/release_notes.md
```
Comment (mariaschuld) [inline doc/index.rst]: Do you want this to appear in the "Developers" section in the docs? It does not work in my build, there are only the original three sections...

File: doc/conf.py (hunk 1 above)
Comment (josh146) [reply doc/conf.py]: Added it to `doc/requirements.txt`!

File: doc/index.rst (hunk 2 above)
Comment (josh146) [reply doc/index.rst]: Oh that's weird, it appears in mine :thinking: Does it work if you `make clean && make docs`?

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -202,55 +252,55 @@ Oluwatobi Ogunbayo, Maria Schuld, Antal Száva.
...
-  [#374](https://github.com/XanaduAI/pennylane/pull/374)
+  [(#374)](https://github.com/XanaduAI/pennylane/pull/374)
 
 * The `CircuitGraph` class is updated to deal with `Operation` instances directly.
-  [#344](https://github.com/XanaduAI/pennylane/pull/344)
+  [(#344)](https://github.com/XanaduAI/pennylane/pull/344)
 
 * Comprehensive gradient tests have been added for the interfaces.
-  [#381](https://github.com/XanaduAI/pennylane/pull/381
+  [(#381)](https://github.com/XanaduAI/pennylane/pull/381
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
  [(#381)](https://github.com/XanaduAI/pennylane/pull/381)
```

File: .github/CHANGELOG.md [hunk 4]
```diff
@@ -632,15 +682,15 @@ Josh Izaac and Nathan Killoran.
 
 # Release 0.2.0
 
-### New features since last release
+<h3>New features since last release</h3>
 
 * Added the `Identity` expectation value for both CV and qubit models (#135)
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
* Added the `Identity` expectation value for both CV and qubit models [(#135)](https://github.com/XanaduAI/pennylane/pull/135)
```

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -632,15 +682,15 @@ Josh Izaac and Nathan Killoran.
...
 * Added the `templates.py` submodule, containing some commonly used QML models to be used as ansatz in QNodes (#133)
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
* Added the `templates.py` submodule, containing some commonly used QML models to be used as ansatz in QNodes [(#133)](https://github.com/XanaduAI/pennylane/pull/133)
```

File: .github/CHANGELOG.md [hunk 6]
```diff
@@ -632,15 +682,15 @@ Josh Izaac and Nathan Killoran.
...
 * Added the `qml.Interferometer` CV operation (#152)
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
* Added the `qml.Interferometer` CV operation [(#152)](https://github.com/XanaduAI/pennylane/pull/152)
```

File: .github/CHANGELOG.md [hunk 7]
```diff
@@ -632,15 +682,15 @@ Josh Izaac and Nathan Killoran.
...
 * Wires are now supported as free QNode parameters (#151)
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
* Wires are now supported as free QNode parameters [(#151)](https://github.com/XanaduAI/pennylane/pull/151)
```

File: .github/CHANGELOG.md [hunk 8]
```diff
@@ -632,15 +682,15 @@ Josh Izaac and Nathan Killoran.
...
 * Added ability to update stepsizes of the optimizers (#159)
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
* Added ability to update stepsizes of the optimizers [(#159)](https://github.com/XanaduAI/pennylane/pull/159)
```

File: .github/CHANGELOG.md [hunk 9]
```diff
@@ -632,15 +682,15 @@ Josh Izaac and Nathan Killoran.
...
 
-### Improvements
+<h3>Improvements</h3>
 
 * Removed use of hardcoded values in the optimizers, made them parameters (see #131 and #132)
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
* Removed use of hardcoded values in the optimizers, made them parameters (see [#131](https://github.com/XanaduAI/pennylane/pull/131) and [#132](https://github.com/XanaduAI/pennylane/pull/132))
```

File: .github/CHANGELOG.md [hunk 10]
```diff
@@ -650,13 +700,13 @@ Josh Izaac and Nathan Killoran.
 * CV analytic gradient calculation is now more robust, allowing for operations which may not themselves be differentiated, but have a
```
Comment (antalszava) [inline .github/CHANGELOG.md]: There are two prior links missing I was unable to edit in the PR

File: .github/CHANGELOG.md [hunk 11]
```diff
@@ -650,13 +700,13 @@ Josh Izaac and Nathan Killoran.
...
   well defined `_heisenberg_rep` method, and so may succeed operations that are analytically differentiable (#152)
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
  well defined `_heisenberg_rep` method, and so may succeed operations that are analytically differentiable [(#152)](https://github.com/XanaduAI/pennylane/pull/152)
```

File: doc/index.rst (hunk 2 above)
Comment (mariaschuld) [reply doc/index.rst]: Yes, that worked. I also used `make html` accidentally, as in the qml repo. Not sure if that makes a difference.

**Comment by mariaschuld** [comment]
Looks very nice with the extra pictures and code!

File: .github/CHANGELOG.md [hunk 12]
```diff
@@ -650,13 +700,13 @@ Josh Izaac and Nathan Killoran.
...
 
-### Bug fixes
+<h3>Bug fixes</h3>
 
 * Fixed a bug where the variational classifier example was not batching when learning parity (see #128 and #129)
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
* Fixed a bug where the variational classifier example was not batching when learning parity (see [#128](https://github.com/XanaduAI/pennylane/pull/128) and [#129](https://github.com/XanaduAI/pennylane/pull/129))
```

File: .github/CHANGELOG.md [hunk 13]
```diff
@@ -650,13 +700,13 @@ Josh Izaac and Nathan Killoran.
...
 * Fixed an inconsistency where some initial state operations were documented as accepting complex parameters - all operations
   now accept real values (#146)
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
  now accept real values [(#146)](https://github.com/XanaduAI/pennylane/pull/146)
```

File: doc/index.rst (hunk 2 above)
Comment (antalszava) [reply doc/index.rst]: It appears for me, maybe that will indeed be solved by a clean

File: .github/CHANGELOG.md [hunk 14]
```diff
@@ -91,107 +141,107 @@ Soran Jahangiri, Zeyue Niu, Maria Schuld, Antal Száva
...
+  [(#427)](https://github.com/XanaduAI/pennylane/pull/427)
 
 * Custom padding constant in `AmplitudeEmbedding` is supported (see 'Breaking changes'.)
-  [#419](https://github.com/XanaduAI/pennylane/pull/419)
+  [(#419)](https://github.com/XanaduAI/pennylane/pull/419)
 
-### Breaking changes
+<h3>Breaking changes</h3>
 
 * The ``pad`` parameter in `AmplitudeEmbedding()`` is now either ``None`` (no automatic padding), or a
```
Comment (mariaschuld) [inline .github/CHANGELOG.md]:
```suggestion
* The ``pad`` parameter in `AmplitudeEmbedding()` is now either `None` (no automatic padding), or a
```

Comment (mariaschuld) [reply .github/CHANGELOG.md]: My bad...

Comment (josh146) [reply .github/CHANGELOG.md]: Spotted another one :laughing: 
```suggestion
* The `pad` parameter in `AmplitudeEmbedding()`` is now either `None` (no automatic padding), or a
```

File: .github/CHANGELOG.md (hunk 3 above)
Comment (josh146) [reply .github/CHANGELOG.md]: Oh, good catch. I used regex, it obviously wasn't a good enough match!

File: .github/CHANGELOG.md [hunk 15]
```diff
@@ -19,403 +19,445 @@
...
   `Device.supports_observable` and `Device.supports_operation`.
   Both methods can be called with string arguments (`dev.supports_observable('PauliX')`) and
   class arguments (`dev.supports_observable(qml.PauliX)`).
-  [#276](https://github.com/XanaduAI/pennylane/pull/276)
+  [(#276)](https://github.com/XanaduAI/pennylane/pull/276)
 
 * The following CV observables were renamed to comply with the new Operation/Observable
   scheme: `MeanPhoton` to `NumberOperator`, `Homodyne` to `QuadOperator` and `NumberState` to `FockStateProjector`.
-  [#243](https://github.com/XanaduAI/pennylane/pull/254)
+  [(#243)](https://github.com/XanaduAI/pennylane/pull/254)
```
Comment (antalszava) [inline .github/CHANGELOG.md]: Here the #243 seems to be a closed one and #254 might be the one that is wanted to be linked (not sure though)

Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
  [(#254)](https://github.com/XanaduAI/pennylane/pull/254)
```

Comment (antalszava) [reply .github/CHANGELOG.md]: (Just left a suggestion assuming that it is correct)

File: .github/CHANGELOG.md [hunk 16]
```diff
@@ -636,31 +676,31 @@ Josh Izaac and Nathan Killoran.
...
+* Added ability to update stepsizes of the optimizers [(#159)](https://github.com/XanaduAI/pennylane/pull/159)
 
-### Improvements
+<h3>Improvements</h3>
 
-* Removed use of hardcoded values in the optimizers, made them parameters (see #131 and #132)
+* Removed use of hardcoded values in the optimizers, made them parameters (see [#131](https://github.com/XanaduAI/pennylane/pull/131) and [#132](https://github.com/XanaduAI/pennylane/pull/132))
 * Created the new `PlaceholderExpectation`, to be used when both CV and qubit expval modules contain expectations with the same name
 * Provide the plugins a way to view the operation queue _before_ applying operations. This allows for on-the-fly modifications of
   the queue, allowing hardware-based plugins to support the full range of qubit expectation values. (#143)
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
  the queue, allowing hardware-based plugins to support the full range of qubit expectation values. [(#143)](https://github.com/XanaduAI/pennylane/pull/143)
```

File: .github/CHANGELOG.md [hunk 17]
```diff
@@ -636,31 +676,31 @@ Josh Izaac and Nathan Killoran.
...
 * QNode return values now support _any_ form of sequence, such as lists, sets, etc. (#144)
```
Comment (antalszava) [inline .github/CHANGELOG.md]:
```suggestion
* QNode return values now support _any_ form of sequence, such as lists, sets, etc. [(#144)](https://github.com/XanaduAI/pennylane/pull/144)
```

File: doc/index.rst [hunk 18]
```diff
@@ -145,16 +145,22 @@ If you are doing research using PennyLane, please cite
...
 - **Source Code:** https://github.com/XanaduAI/PennyLane
 - **Issue Tracker:** https://github.com/XanaduAI/PennyLane/issues
 
 If you are having issues, please let us know by posting the issue on our GitHub issue tracker.
 
-We also have a `PennyLane discussion forum <https://discuss.pennylane.ai>`_—come join the
-discussion and chat with our PennyLane team.
+We encourage contributions — simply fork the PennyLane repository, and then make a
+`pull request <https://help.github.com/articles/about-pull-requests/>`_ containing
+your contribution. All contributers to PennyLane will be listed as authors on the releases.
```
Comment (antalszava) [inline doc/index.rst]: Just a quick question, how come this was moved from ``research.rst`` (which was removed)?

**Pull request merged**